.dockerignore

```
@@ -1,11 +1,73 @@
+# =============================================================================
+# Docker ignore file for GeospatialAnalyzer
+# Prevents unnecessary files from being sent to Docker build context
+# =============================================================================
+
+# Docker and container-related files
 .dockerignore
+Dockerfile*
+docker-compose*.yml
+.docker/
+
+# Version control
+.git/
+.gitignore
+.gitattributes
+
+# Documentation and development files
 documentation/
+README*.md
+*.md
+!swagger-description*.md
+
+# Dependencies (will be installed during build)
 node_modules/
-**/dist
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+pnpm-debug.log*
```

Could be simplified to (especially since we only use `pnpm`):
```suggestion
*.log*
```